### PR TITLE
Harvest: Better synchronisation between components in KonnectorModal

### PR DESCRIPTION
**Pull request by gregorylegarec**
This PR make all the components in KonnectorModal up to date when a job is launched, succeed or throw error.

* AccountErrorInfo si displayed if TriggerManager get errors
* Both TriggerManager and LaunchTriggerCard are rendered as running when a job is launched
* DeleteAccountCard is disabled when a job is launched

As this PR is almost an UX quickfix, it will be followed by another one unifying prop names like running/submitting/isJobRunning and error/konnectorJobError

### CODE REVIEW

File: packages/cozy-harvest-lib/src/components/KonnectorModal.jsx [hunk 1]
```diff
@@ -90,8 +91,13 @@ export class KonnectorModal extends PureComponent {
...
   }
 
-  handleKonnectorJobError(trigger) {
-    this.setState({ isJobRunning: false, trigger })
+  async handleKonnectorJobError() {
+    const { fetchTrigger } = this.props
+    const { trigger } = this.state
+    this.setState({
+      isJobRunning: false,
+      trigger: await fetchTrigger(trigger._id)
```
Comment (ptbrowne) [inline packages/cozy-harvest-lib/src/components/KonnectorModal.jsx]: If the fetchTrigger does not succeed, the `isJobRunning` state attribute is not correctly set. I think the update of `trigger` and setting `isJobRunning` should be decoupled.

File: packages/cozy-harvest-lib/src/components/TriggerManager.jsx [hunk 2]
```diff
@@ -235,18 +239,27 @@ export class TriggerManager extends Component {
...
     this.jobWatcher = watchKonnectorJob(await launchTrigger(trigger))
     this.jobWatcher.on('error', this.handleError)
-    this.jobWatcher.on('loginSuccess', () =>
-      this.handleSuccess(onLoginSuccess, [trigger])
-    )
-    this.jobWatcher.on('success', () =>
-      this.handleSuccess(onSuccess, [trigger])
-    )
+    this.jobWatcher.on('loginSuccess', () => this.handleSuccess(onLoginSuccess))
+    this.jobWatcher.on('success', () => this.handleSuccess(onSuccess))
```
Comment (drazik) [inline packages/cozy-harvest-lib/src/components/TriggerManager.jsx]: A watcher seems to be created on every `launch` call, and the attached events don't seem to be detached. Shouldn't we detach it after error or success?

File: packages/cozy-harvest-lib/src/components/cards/DeleteAccountCard.jsx [hunk 3]
```diff
@@ -26,6 +27,11 @@ export class DeleteAccountCard extends PureComponent {
 
 DeleteAccountCard.propTypes = {
   ...DeleteAccountButton.propTypes,
+  /**
+   * Indicates if the card should be disabled
+   * @type {boolean}
+   */
+  disabled: PropTypes.func,
```
Comment (drazik) [inline packages/cozy-harvest-lib/src/components/cards/DeleteAccountCard.jsx]: It looks like it should be a `PropTypes.bool`

File: packages/cozy-harvest-lib/src/components/TriggerManager.jsx (hunk 2 above)
Comment (gregorylegarec) [reply packages/cozy-harvest-lib/src/components/TriggerManager.jsx]: We could. Actually each event is assumed to occur once : there will be no `error` after `success` and no `success` after `error`.

By the way, we still want to merge this part with the KonnectorJob model. Are you ok to do this in a next PR ?

File: packages/cozy-harvest-lib/src/components/cards/DeleteAccountCard.jsx (hunk 3 above)
Comment (gregorylegarec) [reply packages/cozy-harvest-lib/src/components/cards/DeleteAccountCard.jsx]: Well spotted ! thanks.

File: packages/cozy-harvest-lib/src/components/KonnectorModal.jsx (hunk 1 above)
Comment (gregorylegarec) [reply packages/cozy-harvest-lib/src/components/KonnectorModal.jsx]: Lazy me :)

Comment (gregorylegarec) [reply packages/cozy-harvest-lib/src/components/KonnectorModal.jsx]: BTW I am not happy with this pattern. I guess something is wrong in the conception.

File: packages/cozy-harvest-lib/src/components/TriggerManager.jsx (hunk 2 above)
Comment (drazik) [reply packages/cozy-harvest-lib/src/components/TriggerManager.jsx]: I was also thinking about potential memory leak. Maybe add a TODO comment if you can't fix this now.

Comment (gregorylegarec) [reply packages/cozy-harvest-lib/src/components/TriggerManager.jsx]: https://github.com/cozy/cozy-libs/issues/570 🙏

**Pull request merged**